aderyn_core/src/detect/low/state_variables_not_logged_in_event.rs

```diff
@@ -0,0 +1,244 @@
+        browser::{ExtractEmitStatements, GetImmediateChildren},
+        workspace_context::WorkspaceContext,
+    },
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use crate::{capture, context::browser::ExtractAssignments};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct StateVariableNotLoggedInEventDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for StateVariableNotLoggedInEventDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        /*
+         * How this detector works:
+         * Create a list of all state variables (this includes structs but not their subfields)
+         * Iterates through all functions
+         * Searches each function for assignments
+         * For each assignment found check if the left hand side of the assignment is a state variable using prior created list
+         * If LHS is a state variable then look at all emitted events in the same function
+         * Check if any of the parameters of found events match the state variable
+         * If not, raise an issue
+         */
+
+        // All the state variables, set at the beginning of the detect Function
+
+        let mutable_state_variables: HashMap<i64, VariableDeclaration> = context
+            .variable_declarations()
+            .iter()
+            .filter_map(|&var_decl| {
+                if !var_decl.constant
+                    && matches!(var_decl.mutability(), Some(Mutability::Mutable))
+                    && var_decl.state_variable
+                {
+                    Some((var_decl.id, var_decl.clone())) // clone the VariableDeclaration.
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        for function in context
+            .function_definitions()
+            .into_iter()
+            .filter(|function| !function.is_constructor)
+        {
+            for assignment in ExtractAssignments::from(function).extracted.into_iter() {
+                //if we use an operator such as += we do not want to check for event args with RHS
+                let check_rhs = assignment.operator == "=";
+                let left_hand_side = assignment.left_hand_side.as_ref();
+                let right_hand_side = assignment.right_hand_side.as_ref();
+                let left_identifier_id = return_identifier_referenced_dec(left_hand_side.clone());
+                let right_identifier_name_result = return_identifier_name(right_hand_side.clone());
+                let left_identifier_name_result = return_identifier_name(left_hand_side.clone());
+                let referenced_id = left_identifier_id?;
+                let right_identifier_name = right_identifier_name_result.unwrap();
```

Comment:
If you run `cargo run -- tests/contract-playground`, this line throws an error trying to unwrap @kree-dotcom .

Can you take a look please?

Reply:
Sorry I didn't see the unwrap still present, I also hadn't finished fixing edge cases on the detector, I should have mentioned that here.